### DISCUSSION TITLE
Service worker / offline storage

### DISCUSSION THREAD

**Discussion by mattmuller** [react-native-webview/react-native-webview#3048] [Q&A]
Hi, is it possible to use a service worker for offline storage with this component please?

**Comment by arnaudambro** [comment]
hey there
for those who want to make a SPA working offline - not talking about service worker here, just a SPA working local first and offline
the problem with this lib is that it would check the network before rendering anything, and would render an error if no internet is here

so I went for this solution, that works :
- installed @dr.pogodin/react-native-fs @dr.pogodin/react-native-static-server @dr.pogodin/react-native-webview (which is a fork of that lib)
- download all the assets for the website (html, js, css) - you (or an AI) can create a manifest of the files to download at compile time, accessible at `https://your-website.com/spa-manifest.json` for example, and then fetch all the files one by one in your app, save them in file system
- find a system to check if you have the latest version - easy by stringifying the manifest above
- never serve the url in the webview, always serve the static files you downloaded

and you're done, you have your app working, inline and offline !
